src/docfx/lib/path/PackageUrl.cs

```diff
@@ -77,9 +72,25 @@ private static (string remote, string refspec) SplitGitUrl(string remoteUrl)
 
             path = path.TrimEnd('/', '\\');
             var hasRefSpec = !string.IsNullOrEmpty(fragment) && fragment.Length > 1;
-            var refspec = hasRefSpec ? fragment.Substring(1) : "master";
 
-            return (path, refspec);
+            return (path, hasRefSpec ? fragment.Substring(1) : default);
+        }
+
+        [OnDeserialized]
+        private void OnDeserialized(StreamingContext context)
+        {
+            if (UrlUtility.IsHttp(Url))
+            {
+                Type = PackageType.Git;
+                var (url, branch) = SplitGitUrl(Url);
+                Url = url;
+                Branch = branch ?? Branch;
```

Comment:
it's a little inconsistent, the user specified `Branch` should has higher priority than branch in url?